content/blog/how-to-locate-elements-in-playwright.md

```diff
@@ -0,0 +1,183 @@
+
+If you are not using the VS Code extension Playwright comes with a standalone test generator called [Codegen](https://playwright.dev/docs/test generator-intro).
+
+```bash
+npx playwright codegen
+```
+
+![Codegen](https://res.cloudinary.com/debsobrien/image/upload/v1669647195/debbie.codes/blog/2022/codegen_qykizt.png)
+
+### Pick a locator
+
+You can pick a locator by clicking the [pick locator](https://playwright.dev/docs/next/getting-started-vscode#picking-a-locator) button from the testing sidebar when using the VS Code Extension. 
+
+![pick locator from vs code highlighting the locator in browser window](https://res.cloudinary.com/debsobrien/image/upload/v1669817100/debbie.codes/blog/2022/pick-locator-vscode_xrcdkc.png)
+
+When you hover over any element in the browser window the locator for that element will be highlighted under your cursor. If you click on the element the locator will appear in the 'pick locator' box. You can then copy it to your clipboard and paste it into your test.
+
+You can also pick a locator when using the standalone test generator by clicking the explore button.
+
+```bash
+npx playwright open
+```
+
+![test generator generating a getByRole locator for a link](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-link_hz7icx.png)
+
+### Chaining locators
+
+If your site contains duplicate links, for example two about links, one in the header and one in the footer, then the test generator will create a unique locator for each one. It does this by looking at the next accessible ancestor to the link and chains the locators.
+
+```js
+getByRole('navigation').getByRole('link', { name: 'About' })
+```
+
+![test generator highlighting the about link in the header](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-chaining-header_hacas1.png)
+
+
+For the about link in the footer the test generator again looks at the next accessible ancestor to the link and chains the two [`getByRole()`](https://playwright.dev/docs/locators#locate-by-role) locators ensuring that there is only one instance for this locator.
+
+```js
+getByRole('contentinfo').getByRole('link', { name: 'About' })
+```
+
+![test generator highlighting the footer link](https://res.cloudinary.com/debsobrien/image/upload/v1669795104/debbie.codes/blog/2022/locator-chaining-footer_ogczv7.png)
+
+### Filtering locators
+
+If the test generator can't give you a unique locator when chaining it will use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is unique. For example we might have two lists that contain topics for blog posts. The first list is used to filter the blog posts and the second list is used to display the topics for each post. The test generator will use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is unique.
```

Comment:
```suggestion
If the test generator can't give you a unique locator through chaining it will use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is unique. For example, we might have two lists that contain topics for blog posts. The first list is used to filter the blog posts and the second list is used to display the topics for each post. The test generator will use the [`filter()`](https://playwright.dev/docs/locators#filtering-locators) method to ensure the locator is unique.
```